### PR TITLE
feat(contracts): Adds an more tech related information

### PR DESCRIPTION
**Pull request by Reecepbcups**
- Adds me to contributors page + twitter

- How to do a multi message Tx (bankSend, mint NFTs, etc) + script
- how do I compile a contract?
- How do I deploy a contract to juno?
- How to query a contract vias API / CLI?
- how can you query the price of a token and display on app?
- How do I query all the txs from a specific address?
- How do I send a token to another chain via an api call in my application? (TS Example)
- How do I send a token to a contract via an api call in my application?
- SNAPSHOTS (CW20 & NATIVE)

### CODE REVIEW

File: developer-guides/cosmwasm-contracts/compile-a-contract.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Compile a Contract
+
+[This guide is from the official CosmWasm/rust-optimizer repository](https://github.com/CosmWasm/rust-optimizer). This method of compiling the contract will optimize the final build so that it reduces gas consumption.
+
+Example contracts for this can be found at [https://github.com/CosmWasm/cw-examples](https://github.com/CosmWasm/cw-examples). This repository requires the "Multiple Contract Repository" section.
+
+## Single Contract Repository
+
+The easiest way is to simply use the [published docker image](https://hub.docker.com/r/cosmwasm/rust-optimizer). You must set the local path to the smart contract you wish to compile and it will produce an `artifacts` directory with `<crate_name>.wasm` and `contracts.txt` containing the hashes. This is just one file.
+
```
Comment (Highlander-maker) [inline developer-guides/cosmwasm-contracts/compile-a-contract.md:14]: Section before here on an example of a project. Naming convention. "How to create your own To-Do-List" 

Something like ```cargo generate --git https://github.com/CosmWasm/cw-template.git --name PROJECT_NAME``` if i remember from my old times messing with CW. Maybe a whole new section on Create your own project using CW?

File: developer-guides/cosmwasm-contracts/compile-a-contract.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+```
+docker run --rm -v "$(pwd)":/code \
```
Comment (Highlander-maker) [inline developer-guides/cosmwasm-contracts/compile-a-contract.md:16]: Not sure if my ENV is good im running a M2 chip. 

im seeing this... 

```cw-contracts/contracts/cw-to-do-list is 📦 v0.2.0 via 🦀 v1.66.0
❯ docker run --rm -v "$(pwd)":/code \
  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
  cosmwasm/rust-optimizer:0.12.11
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Info: RUSTC_WRAPPER=sccache
Info: sccache stats before build
Compile requests                      0
Compile requests executed             0
Cache hits                            0
Cache misses                          0
Cache timeouts                        0
Cache read errors                     0
Forced recaches                       0
Cache write errors                    0
Compilation failures                  0
Cache errors                          0
Non-cacheable compilations            0
Non-cacheable calls                   0
Non-compilation calls                 0
Unsupported compiler calls            0
Average cache write               0.000 s
Average cache read miss           0.000 s
Average cache read hit            0.000 s
Failed distributed compilations       0
Cache location                  Local disk: "/root/.cache/sccache"
Cache size                            0 bytes
Max cache size                       10 GiB
Building contract in /code ...
    Updating crates.io index
Killed```

**Review by Highlander-maker — COMMENTED** [review]
I have made a start on the process on this @Reecepbcups. ill leave it tonight and work more tomorrow. Im not sure i have ENV set up correctly. I know i had some issues in the past with my set up. Not sure. But ill continue the review once i get through this first step of compiling.

File: developer-guides/miscellaneous-guides/multi-message-transaction.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      "gas_limit": "200000",
+      "payer": "",
+      "granter": ""
+    }
+  },
+  "signatures": []
+}
+```
+
+Here are are interested in the body -> messages section. In this array, we can put any amount of actions (within the gas limitation) for this transaction to execute with a single execute. So if I wanted to send another amount to a different address at the same time as well, we can manually alter the array of messages to add this new message like so
```
Comment (Highlander-maker) [inline developer-guides/miscellaneous-guides/multi-message-transaction.md]: Typo - Here are are  

Here we are*

File: developer-guides/miscellaneous-guides/multi-message-transaction.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
```
Comment (Highlander-maker) [inline developer-guides/miscellaneous-guides/multi-message-transaction.md]: Maybe show to manually add to the array for example something like 

```Create a file txs.json containing the JSON array:``` 

i created this on uni-6 

```{
  "body": {
    "messages": [
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "juno1p4ar7sytjgjhhcvfnf298ytgklxrgnx4m0zwm2",
        "to_address": "juno1xwazl8ftks4gn00y5x3c47auquc62ssune9ppv",
        "amount": [
          {
            "denom": "ujunox",
            "amount": "1000000"
          }
        ]
      },
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "juno1p4ar7sytjgjhhcvfnf298ytgklxrgnx4m0zwm2",
        "to_address": "juno15twk6xu5rnrrlnf7c5zy92gvykcs4h5ucxzzqq",
        "amount": [
          {
            "denom": "ujunox",
            "amount": "5000000"
          }
        ]
      }
    ],
    "memo": "",
    "timeout_height": "0",
    ...
  }``` 

named it as ```multi_tx_json``` 

i then tried to run ```junod tx broadcast multi_tx.json --chain-id uni-6 --from Highlander --yes --fees 5000ujunox | jq``` 

and get a parse error ```Error: invalid character '.' looking for beginning of object key string: tx parse error
Usage:``` 

Would love your opinion on if im doing this the right way or if im missing something else here. :)

**Review by Highlander-maker — COMMENTED** [review]
Working through * [Multi Message Transaction](developer-guides/miscellaneous-guides/multi-message-transaction.md). 

Really cool stuff. Im attempting to work through the multi tx and hitting some json issues clearly by the level of coding experience i have is the issue. But would appreciate any tips or feedback on my thinking here @Reecepbcups

File: developer-guides/cosmwasm-contracts/compile-a-contract.md (hunk 1 above)
Comment (Reecepbcups) [reply developer-guides/cosmwasm-contracts/compile-a-contract.md:14]: yea this is less about creating and more so how to compile and deploy. I will link to book.cosmwasm.com in the compile stage. This is less about full help & more so just information is that is valuable

File: developer-guides/cosmwasm-contracts/compile-a-contract.md (hunk 2 above)
Comment (Reecepbcups) [reply developer-guides/cosmwasm-contracts/compile-a-contract.md:16]: ahh yea mac has another compiler image, will write in thanks

File: developer-guides/miscellaneous-guides/multi-message-transaction.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+By default, you can only submit 1 transaction per block. In doing this, it will increase your wallets account sequence by one. If you try to submit multiple transactions in the same block, it will error with a account sequence mismatch error. This means there is a pending transaction for your account already in the on chain pending pool (mempool of transactions). Once this transaction goes through and all messages are successfully executed, you can submit another transaction in the following block.
+
+## Multi Message Transaction (bank)
+
+To generate a multiple message transaction, you will need to slight scripting knoladge to assist for your needs. This works for any module in Juno, and you can add as many different messages to the same transaction within the gas limit. (Gas is a deterministic number describing the amount of computer a given action takes on a node).
+
+First, you need to generate a default single message transaction from the command line like so
+
+```sh
+junod tx bank send juno16g2r... juno1t8e... 1000000ujuno --generate-only
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply developer-guides/miscellaneous-guides/multi-message-transaction.md]: good catch, yes

File: developer-guides/miscellaneous-guides/multi-message-transaction.md (hunk 4 above)
Comment (Reecepbcups) [reply developer-guides/miscellaneous-guides/multi-message-transaction.md]: Ahh yes I cut out the bottom part so the ...'s there should not be used. Will give a full copy here

**Comment by Reecepbcups** [comment]
@Highlander-maker The issue was actually with me cutting out the JSON message too early, so it was invalid JSON in my example. I fixed it and should be good now (so long as the from_address is your own address so you can sign correctly)

It does take a bit more programming understanding to really get its full benefit, but it is found no where. and this is the developer section

**Review by Highlander-maker — COMMENTED** [review]
More to come, but left some comments. Great work this @Reecepbcups

File: developer-guides/miscellaneous/multi-message-transaction.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+    "timeout_height": "0",
+    "extension_options": [],
+    "non_critical_extension_options": []
+  },
+  "auth_info": {
+    "signer_infos": [],
+    "fee": {
+      "amount": [
+          {
+              "denom": "ujunox",
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply developer-guides/miscellaneous/multi-message-transaction.md:57]: I use uni-6 as the chain-id in this example, Ill make note of the difference between ujuno vs ujunox chain-ids

File: developer-guides/miscellaneous/multi-message-transaction.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/cosmwasm.wasm.v1.MsgExecuteContract",
+        "sender": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
+        "contract": "juno1za0uemnhzwkjrqwguy34w45mqdlzfm9hl4s5gp5jtc0e4xvkrwjs6s2rt4",
+        "msg": {
+            "mint": {
+                "token_id": f"1",
```
Comment (Highlander-maker) [inline developer-guides/miscellaneous/multi-message-transaction.md:176]: not sure what f is in > `f"1",` ?

Comment (Reecepbcups) [reply developer-guides/miscellaneous/multi-message-transaction.md:176]: ahh, I had fstrings here from python, removing

**Pull request merged**